docs/API/LorisRESTAPI.md

```diff
@@ -0,0 +1,460 @@
+# Loris Instrument API - v0.0.1
+
+## 1.0 Overview
+
+This document specifies the Loris REST API.
+
+Any request sent to `$LorisRoot/api/$APIVERSION/$API_CALL` will return either a JSON object
+or no data. The Loris API uses standard HTTP error codes and the body of any response will
+either be empty or contain only a JSON object for any request.
+
+For brevity, the `$LorisRoot/api/$APIVERSION` is omitted from the definitions in this
+document. This document specifies $APIVERSION v0.0.1 and it
+MUST be included before the request in all requests.
+
+HTTP GET requests NEVER modify data. PUT, POST or PATCH requests MUST be used to modify
+data as per their definitions in the HTTP/1.1 specification. Any methods not supported
+will respond with a 405 Method Not Allowed response and an appropriate Allow header set (as
+per HTTP documentation.)
+
+PUT requests either create or overwrite all data for a given resource (instrument/
+candidate/visit/etc.) Any fields not explicitly specified in the PUT request are nulled.
+
+PATCH requests are identical to PUT requests, but any fields not explicitly mentioned are
+unmodified from their current value.
+
+All GET requests include an ETag header. If a PUT or PATCH request is sent and it does
+not include an ETag, or the ETag does not match the currently existing ETag for that resource,
+it will result in a 403 Forbidden response. PUT or POST requests used for the creation of resources
+do not require ETags.
+
+DELETE is not supported on any resource defined in this API.
+
+# 1.1 Authentication
+
```

Comment:
May be worth stating this is not yet implemented and to use regular login instead